process/coordinator/process.go

```diff
@@ -400,6 +400,10 @@ func (tc *transactionCoordinator) ProcessBlockTransaction(
 		return process.ErrNilBlockBody
 	}
 
+	if tc.isMaxBlockSizeReached(body) {
```

Comment:
as seen in the logs, calling this here will not work as expected because the transactions are not gathered in preprocessors internal caches (yet). They will be added later, after the actual processing. Figuring out a fix for this, also.

Reply:
I don't need them to be created in this case as they are in block body already so I can add them form there as I need only their count. The proposer sent them. This is why inside this method I checked only the mbs with TxBlock type, and inside them I searched only for txs which are cross shard from me, to find some SC calls if they are, to can increase the number of additional txs correctly.

Reply:
Also the receipts and scr from self to self are not included in body. So I don't need to wait for them to be created.

Reply:
Bad explaining on my side: the problem lies in the fact that preprocessor fetch the txs (as structures) from cache only during the actual processing of the miniblock. After that, they will be stored inside the internal cache so they could be written in storage. Calling `tc.GetAllCurrentUsedTxs(blockType)` on line 1013 will yield empty maps.

Reply:
The maps are already populated because the prepocessors already computed missing and existing txs.